tree/treeplayer/inc/ROOT/TDataFrameInterface.hxx

```diff
@@ -1066,26 +1067,33 @@ protected:
       }
 
       {
-         std::unique_ptr<TFile> ofile(TFile::Open(filename.c_str(), "RECREATE"));
-         TTree t(treename.c_str(), treename.c_str());
+         auto df = GetDataFrameChecked();
+         unsigned int nSlots = df->GetNSlots();
+         TBufferMerger merger(filename.c_str(), "RECREATE");
+         std::vector<std::shared_ptr<TBufferMergerFile>> files(nSlots);
+         std::vector<std::shared_ptr<TTree>> trees(nSlots);
```

Comment:
I am not 'finding' the pointer where the shared_ptr are copied/shared.   Wouldn't this work the same with unique_ptr?